doc/doxygen/CMakeLists.txt

```diff
@@ -119,6 +119,26 @@ CONFIGURE_FILE(
   )
 
 
+# Figure out the last copyright date of any of the deal.II source
+# files. We will use this then to set the copyright date of the
+# doxygen-generated HTML files.
+SET(_last_year "1997")
+FILE(GLOB _source_files
+     "${CMAKE_SOURCE_DIR}/source/*/*.cc" 
+     "${CMAKE_SOURCE_DIR}/include/deal.II/*/*.h" 
+     "${CMAKE_SOURCE_DIR}/examples/*/*.cc")
+FOREACH (_source ${_source_files})
+  FILE(READ "${_source}" contents)
+  STRING(REGEX MATCH "Copyright .C. +([0-9]+ - )?([0-9]+)" _copyright ${contents})
+  STRING(REGEX MATCH "[0-9]+\$" _year ${_copyright})
```

Comment:
sometimes getting
```
CMake Error at doc/doxygen/CMakeLists.txt:133 (STRING):
  STRING sub-command REGEX, mode MATCH needs at least 5 arguments total to
  command.


-- doxygen output will have copyright date 2017
```

maybe because `${_copyright}` is empy for some files?